### PR TITLE
New events related to gift card changes

### PR DESCRIPTION
**Pull request by SzymJ**
I want to merge this change because it adds new subscription base webhooks for gift cards create, update, delete, activate and deactivate events.

New webhook events:

* `GIFT_CARD_CREATED`
* `GIFT_CARD_UPDATED`
* `GIFT_CARD_DELETED`
* `GIFT_STATUS_CHANGED`

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6484 Add gift cards webhook events](https://app.clickup.com/t/2549495/SALEOR-6484)

File: saleor/plugins/webhook/plugin.py [hunk 1]
```diff
@@ -114,6 +115,51 @@ def category_deleted(self, category: "Category", previous_value: None) -> None:
...
+            return previous_value
+        event_type = WebhookEventAsyncType.GIFT_CARD_STATUS_CHANGED
+        if webhooks := get_webhooks_for_event(event_type):
+            payload = {
+                "id": graphene.Node.to_global_id("GiftCard", gift_card.id),
+                "is_active": gift_card.is_active,
+            }
+            trigger_webhooks_async(
+                payload, event_type, webhooks, gift_card, self.requestor
+            )
```
Comment (IKarbowiak) [inline saleor/plugins/webhook/plugin.py:161]: It seems like we can have a generic method for all gift card webhooks, as only `event_type` is changing.

File: saleor/plugins/webhook/tests/subscription_webhooks/fixtures.py [hunk 2]
```diff
@@ -79,6 +79,88 @@ def subscription_category_deleted_webhook(subscription_webhook):
...
+    return subscription_webhook(
+        GIFT_CARD_DELETED_SUBSCRIPTION_QUERY, WebhookEventAsyncType.GIFT_CARD_DELETED
+    )
+
+
+GIFT_CARD_STATUS_CHANGED_SUBSCRIPTION_QUERY = """
+    subscription{
+      event{
+        ...on GiftCardStatusChanged{
+          giftCard{
```
[the remark this answers is not in the document]
Comment (SzymJ) [reply saleor/plugins/webhook/tests/subscription_webhooks/fixtures.py:146]: thanks! I didn't know graphql fragments before

File: saleor/graphql/giftcard/bulk_mutations.py [hunk 3]
```diff
@@ -147,6 +150,11 @@ def assign_gift_card_tags(
         for tag_instance in tags_instances.iterator():
             tag_instance.gift_cards.set(instances)
 
+    @staticmethod
+    def trigger_webhook_events(instances, info):
```
[the remark this answers is not in the document]
Comment (SzymJ) [reply saleor/graphql/giftcard/bulk_mutations.py:154]: `call_gift_card_created_on_plugins` will be ok? @korycins

**Review by IKarbowiak — APPROVED** [review]

File: saleor/webhook/event_types.py [hunk 4]
```diff
@@ -18,6 +19,11 @@ class WebhookEventAsyncType:
     CATEGORY_UPDATED = "category_updated"
     CATEGORY_DELETED = "category_deleted"
 
+    GIFT_CARD_CREATED = "gift_card_created"
+    GIFT_CARD_UPDATED = "gift_card_updated"
+    GIFT_CARD_DELETED = "gift_card_deleted"
+    GIFT_CARD_STATUS_CHANGED = "gift_status_changed"
```
Comment (maarcingebala) [inline saleor/webhook/event_types.py:25]:
```suggestion
    GIFT_CARD_STATUS_CHANGED = "gift_card_status_changed"
```
We should re-generate the schema after this change I guess.

Comment (maarcingebala) [inline saleor/webhook/event_types.py:25]:
```suggestion
    GIFT_CARD_STATUS_CHANGED = "gift_card_status_changed"
```
We should re-generate the schema after this change I guess.

**Review by maarcingebala — APPROVED** [review]

**Review by korycins — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**